### DISCUSSION TITLE
Premission denied for schema public

### DISCUSSION THREAD

**Discussion by Vigintillionn** [prisma/prisma#24319] [Q&A] [answered]
### Question

I have a prisma setup using Postgresql (on my vps running fedora). I created a new user and a database for this user. The user has CREATEDB permissions and full access to the database. Yet I get this error:

```
Error: ERROR: permission denied for schema public
   0: sql_schema_connector::sql_migration_persistence::initialize
           with namespaces=None
             at schema-engine/connectors/sql-schema-connector/src/sql_migration_persistence.rs:14
   1: schema_core::state::ApplyMigrations
             at schema-engine/core/src/state.rs:201
```

I already tried giving it privileges for the schema public but this did not resolve the issues. I can't find any info online about how to solve this issue. Any help is appreciated.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Arch (running on my machine), fedora on the VPS
- Database: Postgress
- Node.js version: v22.2.0

- Prisma version: 5.14.0

**Comment by ludralph** [comment, marked as answer]
Hi @Vigintillionn 👋 

Thank you for raising this question.

The error message "permission denied for schema public" indicates a permission issue with the schema. Did you run this command `GRANT ALL ON SCHEMA public TO <username>;`?

**Reply by Vigintillionn** [reply]
Hello, thanks for the reply. I thought I did, but after running it again it solved my issue. Thanks alot.

**Comment by madhavpujara-at** [comment]
@ludralph we had the same issue, but in our case our db is public when we ran the migrate command it succesfully ran from the local machine and failed when we used is inside the aws ecs  container

**Reply by madhavpujara-at** [reply]
We need to find the root cause of this inconsistent behaviour as after the migrations ran from the local machine the ecs worked correctly , but running migration from local machine , in a ci/cd loop is all the wrong types of fix
